### PR TITLE
Added CIFAR100 to built-in datasets, issue #86.

### PR DESCRIPTION
**Pull request by julianser**
@vdumoulin

### CODE REVIEW

File: fuel/downloaders/cifar100.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from fuel.downloaders.base import default_downloader
+
+
+def cifar100(subparser):
+    """Sets up a subparser to download the CIFAR-100 dataset file.
+
+    The CIFAR-109 dataset file is downloaded from Alex Krizhevsky's
```
Comment (vdumoulin) [inline fuel/downloaders/cifar100.py]: Typo: `CIFAR-109`

File: fuel/downloaders/cifar100.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    website [ALEX].
+
+    .. [ALEX] http://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz
+
+    Parameters
+    ----------
+    subparser : :class:`argparse.ArgumentParser`
+        Subparser handling the `cifar10` command.
```
Comment (vdumoulin) [inline fuel/downloaders/cifar100.py]: Typo: `cifar10`

**Comment by vdumoulin** [comment]
Can you also add unit tests like [here](https://github.com/bartvm/fuel/blob/master/tests/test_cifar10.py)?

**Comment by vdumoulin** [comment]
If my comments are addressed and the Scrutinizer errors are fixed, we'll be good to merge (Travis is under maintenance, I'll manually restart the tests if need be).

**Comment by vdumoulin** [comment]
The current test failure is caused by the dataset not being present in the Travis build bot. You can solve it by adding a block of code similar to [this code block](https://github.com/bartvm/fuel/blob/master/.travis.yml#L44-L52) right below in `.travis.yml`.

Also, in order to prevent the tests from failing if the dataset isn't present, you can add a `skip_if_not_available` statement at the beginning, like [here](https://github.com/bartvm/fuel/blob/master/tests/test_mnist.py#L15).

**Comment by vdumoulin** [comment]
There are still [PEP8 errors](https://scrutinizer-ci.com/g/bartvm/fuel/inspections/90643fdc-8e20-4ff7-96b8-fc9c6c01720f) to fix, but the Travis failures look unrelated to your PR. I would merge your PR as soon as you fix the PEP8 errors and I'll deal with the unrelated test failures afterwards if need be.

**Comment by vdumoulin** [comment]
There's been a slight change in how converters work and how converters and downloaders register to their respective module:

http://fuel.readthedocs.org/en/latest/new_dataset.html#download-code

http://fuel.readthedocs.org/en/latest/new_dataset.html#conversion-code

**Comment by julianser** [comment]
I've made a new (cleaner) PR.

**Comment by dwf** [comment]
FYI there's never really a need to create a new PR, you can just push to
the same remote branch with --force.

**Pull request closed without merging**